File: CPP/CWE-477/2/bad.cpp
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */
#include <cstdlib>
#include <iostream>
#include <strstream>

using namespace std;

int main(int argc, char *argv[])
{
	strstreambuf str; //deprecated by C++98
	strstream str_s;
	str_s << 3.14 << ends;
	return EXIT_SUCCESS;
}

File: C#/CWE-477/2/bad.cs
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */

using System;

namespace BadWeakness
{
	class Bad
	{
		static void Main()
		{
			Deprecated d = new Deprecated();
			d.DeprecatedServerName = "example.com";
			Console.WriteLine(d.ServerName);
		}
	}

	class Deprecated
	{
		private string _server;

		public string ServerName { get { return _server; } set {_server = value; } }
		[ObsoleteAttribute("Use ServerName instead.")]
		public string DeprecatedServerName { get { return _server; } set {_server = value; } }
	}
}
